### PR TITLE
[exporter/awsemf] Add feature gate for default dimension rollup

### PR DESCRIPTION
**Pull request by bryan-aguilar**
**Description:** Add feature gate to change default DimensionRollupOption. This was introduced as a feature gate since it will be a breaking change when switching to Beta. 

**Link to tracking Issue:** #23997

**Testing:** Added unit tests

**Documentation:** Updated readme.

### CODE REVIEW

File: exporter/awsemfexporter/emf_exporter.go [hunk 1]
```diff
@@ -71,6 +71,10 @@ func newEmfExporter(config *Config, set exporter.CreateSettings) (*emfExporter,
 		pusherMap:        map[cwlogs.PusherKey]cwlogs.Pusher{},
 	}
 
+	config.logger.Warn("the default value for value for DimensionRollupOption will be changing to NoDimensionRollup" +
```
Comment (atoulme) [inline exporter/awsemfexporter/emf_exporter.go:74]:
```suggestion
	config.logger.Warn("the default value for DimensionRollupOption will be changing to NoDimensionRollup" +
```

File: exporter/awsemfexporter/emf_exporter_test.go [hunk 2]
```diff
@@ -589,6 +589,12 @@ func TestNewExporterWithMetricDeclarations(t *testing.T) {
 
 	// Test output warning logs
 	expectedLogs := []observer.LoggedEntry{
+		{
+			Entry: zapcore.Entry{Level: zap.WarnLevel, Message: "the default value for value for DimensionRollupOption will be changing to NoDimensionRollup" +
```
Comment (atoulme) [inline exporter/awsemfexporter/emf_exporter_test.go:593]:
```suggestion
			Entry: zapcore.Entry{Level: zap.WarnLevel, Message: "the default value for DimensionRollupOption will be changing to NoDimensionRollup" +
```

**Review by Aneurysm9 — APPROVED** [review]

**Comment by atoulme** [comment]
Please attend to the failing build.

**Comment by bryan-aguilar** [comment]
@atoulme updating main resolved failures.

**Comment by bryan-aguilar** [comment]
@open-telemetry/collector-contrib-maintainer this has been ready for merge for 3 weeks now. The only failing tests that occur are flakes after continually having to update with main branch.

File: exporter/awsemfexporter/factory.go [hunk 3]
```diff
@@ -9,13 +9,19 @@ import (
...
+	"go.opentelemetry.io/collector/featuregate"
 	"go.uber.org/zap"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsemfexporter/internal/metadata"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/awsutil"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/resourcetotelemetry"
 )
 
+var defaultNoRollupfg = featuregate.GlobalRegistry().MustRegister("awsemf.nodimrollupdefault", featuregate.StageAlpha,
+	featuregate.WithRegisterFromVersion("v0.82.0"),
```
Comment (TylerHelmuth) [inline exporter/awsemfexporter/factory.go:21]:
```suggestion
	featuregate.WithRegisterFromVersion("v0.83.0"),
```

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**